### PR TITLE
Introducing OSP7 SCAP Content

### PR DESCRIPTION
**Pull request by shawndwells**
Not ready to merge!

Somehow fudged the Makefile, whereas no OVAL content is being merged in. A `````make clean ; make````` results in the following:

`````
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 9: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}definitions': Missing child element(s). Expected is ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}definition ).
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 10: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}tests': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_test ).
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 11: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}objects': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_object ).
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 12: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}states': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_state ).
OpenSCAP Error: Invalid SCAP Source Datastream (1.2) content in /var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml. [oscap_source.c:205]
make: *** [content] Error 1

`````

I'm not entirely sure why the various OVAL checks are not being pulled in. Any peer review would be welcome!

### CODE REVIEW

**Comment by isimluk** [comment]
I looked into that, first I have found that we need to have the missing directory for jenkins to pass:
```
mkdir -p  OpenStack/input/remediations/bash
touch OpenStack/input/remediations/bash/.gitignore
git add OpenStack/input/remediations/bash/.gitignore 
```

**Comment by isimluk** [comment]
Then, I am getting
```
xsltproc -stringparam ref "nist" -o output/table-rhel7-nistrefs.html transforms/xccdf2table-byref.xslt output/unlinked-rhel7-empty-groups-xccdf.xml
warning: failed to load external entity "transforms/table-style.xslt"
compilation error: file transforms/xccdf2table-byref.xslt line 7 element include
xsl:include : unable to load transforms/table-style.xslt
```
Is it possible, that you have this transformation on your local harddrive?

If not please run something like
```
cp RHEL/7/transforms/table-style.xslt OpenStack/transforms/
git add OpenStack/transforms/table-style.xslt
```

**Comment by isimluk** [comment]
Then I got
```
find input/oval/oval_5.11 -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel7_oval ';'
find: ‘input/oval/oval_5.11’: No such file or directory
Makefile:61: recipe for target 'checks' failed
```
Which will be again fixed in
```
mkdir -p OpenStack/input/oval/oval_5.11
touch OpenStack/input/oval/oval_5.11/.gitignore
git add  OpenStack/input/oval/oval_5.11/.gitignore
```

**Comment by isimluk** [comment]
Then, I have got
```
xsltproc -stringparam overlay "../input/auxiliary/stig_overlay.xml" -o output/unlinked-stig-rhel7-xccdf.xml \
	transforms/xccdf-apply-overlay-stig.xslt \
	output/unlinked-rhel7-xccdf.xml
warning: failed to load external entity "transforms/xccdf-apply-overlay-stig.xslt"
cannot parse transforms/xccdf-apply-overlay-stig.xslt
Makefile:102: recipe for target 'table-stigs' failed
```
which I don't know how to fix. Because I don't know if we want to build these stigs tables for OpenStack. If we want them, we need to have the transforms/xccdf-apply-overlay-stig.xslt in the directory.

**Comment by isimluk** [comment]
I feel like our build system is dungeon with a dragons constantly throwing a fire through the barred windows. Only the strongest characters can hold out to dinner. :-)

I think this should improve overtime, as we brink the makefiles and all the tooling to be the same in each directory. However, please do not be discouraged since then.

**Comment by isimluk** [comment]
So, after I copied the xccdf-apply-overlay-stig.xslt. I get

```
echo "XXXXXX"
XXXXXX
xsltproc -o output/unlinked-rhel7-xccdf-guide.xml transforms/xccdf-removeaux.xslt output/unlinked-rhel7-xccdf.xml
../shared/transforms/cpe_generate.py output/unlinked-rhel7-oval.xml input/oval/platform/rhel7-cpe-dictionary.xml ssg
Traceback (most recent call last):
  File "../shared/transforms/cpe_generate.py", line 117, in <module>
    main()
  File "../shared/transforms/cpe_generate.py", line 106, in main
    cpedicttree = parse_xml_file(cpedictfile)
  File "../shared/transforms/cpe_generate.py", line 19, in parse_xml_file
    with open(xmlfile, 'r') as xml_file:
IOError: [Errno 2] No such file or directory: 'input/oval/platform/rhel7-cpe-dictionary.xml'
Makefile:119: recipe for target 'content' failed
```

So

```
mkdir OpenStack/input/oval/platform
cp RHEL/7/input/oval/platform/rhel7-cpe-dictionary.xml ./OpenStack/input/oval/platform/
git add ..
```

**Comment by isimluk** [comment]
Then, I have got
```
xsltproc -o output/ssg-rhel7-oval.xml transforms/oval-fix-test-attestation-urls.xslt output/ssg-rhel7-oval.xml
warning: failed to load external entity "output/ssg-rhel7-oval.xml"
unable to parse output/ssg-rhel7-oval.xml
```

I got thinking maybe there is no oval content at all. And I am doing it all wrong. :-)

**Comment by isimluk** [comment]
@shawndwells , sorry for the morning confusion in this pr. I have spent some time today understanding the makefiles more in depth. And I have identified some areas that can be improved. I'll work towards that goal. I don't have an interim solution for this PR, but I envision that the Makefiles will be much smaller and efficient in couple weeks.

**Comment by shawndwells** [comment]
Thanks @isimluk. I'll revisit later this week.

**Comment by iankko** [comment]
@shawndwells 
Looked further into this. There are two issues at least:
* for the not included OVALs error / warning message:
```
# cd RHEL-OSP/7
# make clean
rm -f output/*.xml output/*.html output/*.xhtml output/*.pdf  output/*.spec output/*.tar output/*.gz output/*.ini output/*.csv
rm -rf dist/content dist/guide
rm -rf build
# make checks
# Make intermediate build/rhel-osp7_oval directory to hold final list of OVAL checks for rhel-osp7
mkdir -p build/rhel-osp7_oval
# Search ../../shared/oval and input/oval directories to find all product specific OVAL checks,
# which are regular files (not symlinks). Merge the final list into build/rhel-osp7_oval directory
#find ../../shared/oval input/oval -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel-osp7_oval ';'
find input/oval -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel-osp7_oval ';'
# If openscap on the system supports OVAL-5.11 language version, include also OVAL-5.11 checks
# into final list of OVAL checks
find input/oval/oval_5.11 -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel-osp7_oval ';'
# Update requested RHEL-7 OVAL version in ../../config/oval.config to be the 5.11 OVAL language version
sed -i "s/^\(rhel7_oval_version = \).*/\15.11/" ../../config/oval\.config
xmlwf build/rhel-osp7_oval/*.xml
../../shared/transforms/combineovals.py ../../config rhel-osp7 build/rhel-osp7_oval > output/unlinked-rhel-osp7-oval.xml

Notification: Merged 0 OVAL checks into OVAL document.
xmllint --format --output output/unlinked-rhel-osp7-oval.xml output/unlinked-rhel-osp7-oval.xml
```

The issue is ```multi_platform_rhel``` ```<platform>``` value is used for OVAL checks for ```RHEL-OSP/7```:
```
# grep platform input/oval/cinder_file_perms.xml 
        <platform>multi_platform_rhel</platform>
```

This is possible to do. But if you want to do that, it's necessary instruct the build system (namely the ```shared/transforms/combineovals.py``` script to treat ```multi_platform_rhel``` platform as valid also for ```RHEL-OSP/7``` product. In other words whitelist ```multi_platform_rhel``` platform also for the newly defined / introduced ```rhel-osp``` product ID). This can be achieved with the following patch:

<b>Patch 1:</b>

```
$ git diff shared/transforms/combineovals.py
diff --git a/shared/transforms/combineovals.py b/shared/transforms/combineovals.py
index 95ac4c0..c5db262 100755
--- a/shared/transforms/combineovals.py
+++ b/shared/transforms/combineovals.py
@@ -109,7 +109,7 @@ def check_is_applicable_for_product(oval_check_def, product):
     # First test if OVAL check isn't for 'multi_platform_all' or
     # 'multi_platform_' + product
     for mp in multi_platforms:
-        if mp in oval_check_def and product in ['rhel', 'fedora']:
+        if mp in oval_check_def and product in ['rhel', 'fedora', 'rhel-osp']:
             return True
 
     # Current SSG checks aren't unified which element of '<platform>'
```

This build-system behaviour has been introduced because:
* we can't use symbolic links anymore to indicate which OVAL should be merged / included into which particular product,
* therefore we are parsing the content of the provided ```<platform>``` tag within particular ```shorthand``` OVAL definition and behaving as follows:
 * if ```<platform>``` tag is unambiguous, e.g. ```<platform>Red Hat Enterprise Linux 7</platform>```, and product is ```rhel7``` we include that oval into further build stages,
* if ```<platform>``` tag is multi-product one, we do the following:
 * we include OVALs having ```<platform>multi_platform_all</platform>``` into both products ```rhel``` and ```fedora```,
 * we include OVALs having ```<platform>multi_platform_rhel</platform>``` only into ```rhel``` products (where product ID comes from ```Makefile``` for that product), and finally
 * include OVALs having ```<platform>multi_platform_fedora</platform>``` only into ```fedora``` product.

If you want to have a further look for the logic, have a look at ```check_is_applicable_for_product``` routine implementation in ```shared/transforms/combineovals.py```.  This logic is here to put a barrier, not to include ```multi_platform_rhel``` platform-tagged OVALs everywhere (e.g. into ```JRE``` for example).

This has been working so far. But if another / new product wants to be able to use these ```multi_platform_rhel``` platform-tagged OVALs it's ID needs to be added into the whitelist to
```shared/transforms/combineovals.py``` as it's performed by aforementioned <b>Patch 1:</b> above.

Once <b> Patch 1:</b> from the above is applied (IOW 'rhel-osp' product is whitelisted to re-use ```multi_platform_rhel``` platform-tagged OVALs), the newly provided OVAL checks are already included:

```
# cd RHEL-OSP/7
# make clean
rm -f output/*.xml output/*.html output/*.xhtml output/*.pdf  output/*.spec output/*.tar output/*.gz output/*.ini output/*.csv
rm -rf dist/content dist/guide
rm -rf build

# make checks
# Make intermediate build/rhel-osp7_oval directory to hold final list of OVAL checks for rhel-osp7
mkdir -p build/rhel-osp7_oval
# Search ../../shared/oval and input/oval directories to find all product specific OVAL checks,
# which are regular files (not symlinks). Merge the final list into build/rhel-osp7_oval directory
#find ../../shared/oval input/oval -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel-osp7_oval ';'
find input/oval -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel-osp7_oval ';'
# If openscap on the system supports OVAL-5.11 language version, include also OVAL-5.11 checks
# into final list of OVAL checks
find input/oval/oval_5.11 -maxdepth 1 -type f -name *.xml -exec cp {} build/rhel-osp7_oval ';'
# Update requested RHEL-7 OVAL version in ../../config/oval.config to be the 5.11 OVAL language version
sed -i "s/^\(rhel7_oval_version = \).*/\15.11/" ../../config/oval\.config
xmlwf build/rhel-osp7_oval/*.xml
../../shared/transforms/combineovals.py ../../config rhel-osp7 build/rhel-osp7_oval > output/unlinked-rhel-osp7-oval.xml

Notification: Merged 6 OVAL checks into OVAL document.
xmllint --format --output output/unlinked-rhel-osp7-oval.xml output/unlinked-rhel-osp7-oval.xml
```

**Comment by iankko** [comment]
The second issue is related with the following error message, you quote above:
```
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 9: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}definitions': Missing child element(s). Expected is ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}definition ).
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 10: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}tests': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_test, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_test ).
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 11: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}objects': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_object, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_object ).
File '/var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml' line 12: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}states': Missing child element(s). Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_state ).
OpenSCAP Error: Invalid SCAP Source Datastream (1.2) content in /var/www/html/scap-security-guide/RHEL-OSP/7/output/ssg-rhel-osp7-ds.xml. [oscap_source.c:205]
make: *** [content] Error 1
```

Having inspected this, the problem is the empty ```<states/>``` element in the ```output/ssg-rhel-osp7-cpe-oval.xml``` and ```output/unlinked-rhel-osp7-oval.xml```:
```
# grep 'states' output/ssg-rhel-osp7-oval.xml 
  <states/>

# grep 'states' output/unlinked-rhel-osp7-oval.xml 
  <states/>
```
That's actually where all those ```missing child elements``` warnings come from. The produced OVAL isn't valid now:
```
# oscap oval validate output/unlinked-rhel-osp7-oval.xml
```
also goes to that warning. ```unlinked-$(PROD)-oval.xml``` file is generated by ```shared/transforms/cpe_generate.py``` script from ```$(IN)/oval/platform/$(PROD)-cpe-dictionary.xml``` file. Look at ``` input/oval/platform/rhel-osp7-cpe-dictionary.xml ``` file suggests it's referencing another OVAL checks:
* ```installed_OS_is_rhel7```,
* ```installed_OS_is_centos7```,
* ```installed_OS_is_sl7```

and none of these is defined in the ```RHEL-OSP/7``` product. Once e.g. ```installed_OS_is_rhel7``` implementation from ```shared/oval``` is copied into ```RHEL-OSP/7/input/oval``` the content is produced:
```
# cp shared/oval/installed_OS_is_rhel7.xml RHEL-OSP/7/input/oval/
# cd RHEL-OSP/7/
# make clean
# make content
..
oscap ds sds-add output/ssg-rhel-osp7-oval.xml output/ssg-rhel-osp7-ds.xml
../../shared/utils/enable-derivatives.py --enable-centos -i output/ssg-rhel-osp7-xccdf.xml -o output/ssg-centos7-xccdf.xml
../../shared/utils/enable-derivatives.py --enable-centos -i output/ssg-rhel-osp7-ds.xml -o output/ssg-centos7-ds.xml
../../shared/utils/enable-derivatives.py --enable-sl -i output/ssg-rhel-osp7-xccdf.xml -o output/ssg-sl7-xccdf.xml
../../shared/utils/enable-derivatives.py --enable-sl -i output/ssg-rhel-osp7-ds.xml -o output/ssg-sl7-ds.xml
# echo $?
0
```

The natural question might be why e.g. ```RHEL/7``` content works, when there isn't ```installed_OS_is_rhel7.xml``` directly listed in ```RHEL/7/input/oval``` folder. The answer is ```RHEL/7```'s Makefile searches also content of the ```shared/oval``` directory for applicable OVAL checks. And since ```installed_OS_is_rhel7``` OVAL has ```<platform>multi_platform_all</platform>``` tag, it's included during this search also for ```RHEL/7``` product (IOW it is not necessary to explicitly copy it).

But the ```shared/oval``` folder in ```RHEL-OSP/7``` product reads as follows:
```
# cd RHEL-OSP/7
# grep -Hn '$(SHARED_OVAL)' Makefile 
Makefile:63:	# Search $(SHARED_OVAL) and $(IN)/oval directories to find all product specific OVAL checks,
Makefile:65:	#find $(SHARED_OVAL) $(IN)/oval -maxdepth 1 -type f -name *.xml -exec cp {} $(PROD_OVAL) ';'
```

In other words that ```$(SHARED_OVAL)``` content is commented out for ```RHEL-OSP/7``` product (this is expected, since we don't want all existing ```RHEL/6``` and ```RHEL/7```  OVAL checks from the ```shared``` folder to be also applicable / e.g. included also into OVAL for ```RHEL-OSP/7``` product). But if this is the case (we don't want all the ```shared/``` OVALs to apply for ```RHEL-OSP/7```, that ```installed_OS_is_rhel7``` OVAL needs to be copied into ```RHELL-OSP/7/input/oval``` directory, so its content is included into ```output/ssg-rhel-osp7-cpe-oval.xml```, and subsequently by ```shared/transforms/cpe_generate.py``` transformation into ```output/ssg-rhel-osp7-oval.xml```.

**Comment by iankko** [comment]
I realize the previous two comments might seem complex, thus will mention only action items:
* If ```multi_platform_rhel` / fedora``` OVALs should be applicable also to some other products than just ```RHEL/*``` and ```Fedora```, that product ID (coming from ```Makefile```) needs to be whitelisted in ```shared/transforms/combineovals.py``` script (see <b>Patch 1:</b> for details how to do this),

* if ```$(IN)/oval/platform/$(PROD)-cpe-dictionary.xml``` definition references some OVAL and can't use the version from the ```shared/oval``` directory (e.g. doesn't want to search also ```shared/oval``` directory for available OVALs), then OVAL checks referenced within ```$(IN)/oval/platform/$(PROD)-cpe-dictionary.xml``` need to be copied into ```$(PROD)/$(IN)/oval``` directory.

Yet for the ability of the build system to produce e.g. ```unlinked-rhel-osp7-oval.xml``` [ ```unlinked-$(PROD)-oval.xml```] with empty ```<states/>``` element (e.g. that missing child elements warning above), I will file a PR tomorrow to have this corrected (this not to be possible in the future).

**Comment by iankko** [comment]
@shawndwells 
A small nitpick regarding the directory layout yet. Why not to delete the currently existing ```OpenStack``` folder (there's no content at all as far as I can tell), and replace it wit the content of the newly introduced ```RHEL-OSP/7``` folder?

Or if versioned OpenStack layout is necessary, why not to create something like:
```
$ tree OpenStack/
OpenStack/
├── RHELOSP
│   ├── 6
│   └── 7
└── upstream

4 directories, 0 files
```

The issue is this way all the OpenStack related SCAP content could be under one directory.

@redhatrises , @mpreisler , @isimluk Any opinion on the layout -^?

**Comment by shawndwells** [comment]
On 12/14/15 3:33 PM, iankko wrote:
>
> *Patch 1:*
>
> |$ git diff shared/transforms/combineovals.py diff --git 
> a/shared/transforms/combineovals.py 
> b/shared/transforms/combineovals.py index 95ac4c0..c5db262 100755 --- 
> a/shared/transforms/combineovals.py +++ 
> b/shared/transforms/combineovals.py @@ -109,7 +109,7 @@ def 
> check_is_applicable_for_product(oval_check_def, product): # First test 
> if OVAL check isn't for 'multi_platform_all' or # 'multi_platform_' + 
> product for mp in multi_platforms: - if mp in oval_check_def and 
> product in ['rhel', 'fedora']: + if mp in oval_check_def and product 
> in ['rhel', 'fedora', 'rhel-osp']: return True # Current SSG checks 
> aren't unified which element of '<platform>' |
Thank you @iankko! Pushed to branch as 175c406.

**Comment by shawndwells** [comment]
On 12/14/15 4:17 PM, iankko wrote:
>
> @shawndwells <https://github.com/shawndwells>
> A small nitpick regarding the directory layout yet. Why not to delete 
> the currently existing |OpenStack| folder (there's no content at all 
> as far as I can tell), and replace it wit the content of the newly 
> introduced |RHEL-OSP/7| folder?
>
> Or if versioned OpenStack layout is necessary, why not to create 
> something like:
>
> |$ tree OpenStack/ OpenStack/ ├── RHELOSP │   ├── 6 │   └── 7 └── 
> upstream 4 directories, 0 files |
>
> The issue is this way all the OpenStack related SCAP content could be 
> under one directory.
>
> @redhatrises <https://github.com/redhatrises> , @mpreisler 
> <https://github.com/mpreisler> , @isimluk <https://github.com/isimluk> 
> Any opinion on the layout -^?
>

I like OpenStack/${distribution}/${version}

**Comment by shawndwells** [comment]
Jenkins is reporting `````stderr: fatal: Unable to look up github.com (port 9418) (Temporary failure in name resolution)`````, however latest patches rebase against upstream and resolve merge conflicts.

OpenStack content should now build, passes "make clean && make && make validate"

**Comment by shawndwells** [comment]
@iankko  thanks so much for the pointers!

**Comment by isimluk** [comment]
[test this please]

**Comment by iankko** [comment]
@shawndwells 
Fixed the conflicts && merged the changes manually. The ```OpenStack/RHEL-OSP/7/``` content doesn't support OVAL-5.11 yet (this will need to be enhanced in future PRs). but it successfully builds (```make```) and OVAL-5.10 version validates (```make validate```).

We fill fix the rest in the future.

**Comment by shawndwells** [comment]
Thanks @iankko ! With other changesets to Makefiles and other code cleanup, this PR was hard to keep fresh. So far there aren't OVAL checks that *need* 5.11, but yes you are fully correct the content doesn't support yet. Thanks for the review and merge!
